File: types/entity.ts
export type EntityType = "vendor" | "project" | "contract" | "purchase_order";

export interface Entity {
  id: string;
  type: EntityType;
  name: string;
  metadata?: {
    trade?: string;
    contact?: string;
    location?: string;
    [key: string]: any;
  };
  created_at?: string;
}

File: models/entity.py
# Entity model definition matching 03-DATABASE-SCHEMA.md

class Entity:
    def __init__(self, id: str, name: str, entity_type: str, metadata: dict = None):
        self.id = id
        self.name = name
        self.entity_type = entity_type
        self.metadata = metadata or {}
